packages/shared/components/filters/AssetFilter.svelte

```diff
@@ -0,0 +1,32 @@
+<script lang="typescript">
+    import { TogglableButton, FilterModal, FilterItem, Modal } from 'shared/components'
+    import { AssetFilter } from '@core/wallet/interfaces/filter.interface'
+    import { assetFilter } from '@core/wallet/stores/selected-account-assets.store'
+
+    let filter: AssetFilter = { ...$assetFilter }
+
+    let filterActive = false
+    let modal: Modal
+
+    $: filterActive && modal?.toggle()
+
+    function onSetFilters() {
+        modal?.toggle()
+        filterActive = false
+        $assetFilter = filter
+    }
+</script>
+
+<div class="relative">
+    <TogglableButton icon="filter" bind:active={filterActive} />
+
+    {#if filterActive}
+        <div class="absolute right-60 top-30">
+            <FilterModal bind:modal bind:filter {onSetFilters}>
+                {#each Object.keys(filter) as filterItem}
+                    <FilterItem filterItem={filter[filterItem]} />
+                {/each}
+            </FilterModal>
+        </div>
+    {/if}
+</div>
```

Comment:
Looks like this is a duplication of `ActivityFilter`, maybe it's possible to use generics for the types and create a `Filter` oganism that is used by both?

Reply:
Good point, implemented it in the newest commits